gobblin-modules/gobblin-kafka-common/src/main/java/org/apache/gobblin/source/extractor/extract/kafka/workunit/packer/KafkaTopicGroupingWorkUnitPacker.java

```diff
@@ -251,7 +251,7 @@ private void addStatsToWorkUnits(Map<String, List<WorkUnit>> workUnitsByTopic) t
       List<WorkUnit> workUnits = entry.getValue();
       for (WorkUnit workUnit : workUnits) {
         int partitionId = Integer.parseInt(workUnit.getProp(KafkaSource.PARTITION_ID));
-        String topicPartition = new KafkaPartition.Builder().withTopicName(topic).withId(partitionId).toString();
+        String topicPartition = new KafkaPartition.Builder().withTopicName(topic).withId(partitionId).build().toString();
```

Comment:
topicPartition has been declared within both the loops. Can we declare a common variable topicPartition and use it ?

Reply:
are there any unit tests for this change?

Reply:
@abhizen The variable is re-declared in two independent scopes. I don't see the advantage from a code-readability standpoint to have one common declaration. Also, I think it would be good to a unit test for addStatsToWorkUnits(). Would create a separate PR for it.